### ISSUE TITLE
In python3.8, the position of Unexpected Error in the stack trace is strange.

### ISSUE THREAD

**Issue by podhmo** [pallets/jinja#1048]
In python3.8, the position of Unexpected Error in the stack trace is strange.
Access to an undefined variable is on line 3, but according to the error message it is on line 1.

```python
import jinja2

t = """\
1: # access undefined
2: 
3: {{xxx}} # <- accessing undefined variable, here."""

print(jinja2.__version__)
print(jinja2.Template(t, undefined=jinja2.StrictUndefined).render())
```

## Expected Behavior

expected 3rd line.

```
  File "<template>", line 3, in top-level template code
jinja2.exceptions.UndefinedError: 'xxx' is undefined
```

## Actual Behavior

but 1st line.

```
  File "<template>", line 1, in <module>
jinja2.exceptions.UndefinedError: 'xxx' is undefined
```

## Template Code
```jinja
1: # access undefined
2: 
3: {{xxx}} # <- accessing undefined variable, here.
```

## Full Traceback
```pytb
2.10.1
Traceback (most recent call last):
  File "render-with-undefined.py", line 9, in <module>
    print(jinja2.Template(t, undefined=jinja2.StrictUndefined).render())
  File "/usr/local/lib/python3.8/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.8/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.8/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "<template>", line 1, in <module>
jinja2.exceptions.UndefinedError: 'xxx' is undefined
```

## Your Environment

* Python version: 3.8.0b3
* Jinja version: 2.10.1

## Detail Information

https://gist.github.com/podhmo/0308dd8e174c3ec66a451272e4e43939

**Comment by davidism** [comment]
Duplicate of #276 except Python 3.8 changed from reporting the last line to the first line of multiline expressions.

**Comment by davidism** [comment]
Fixed by #1109
